packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -60,14 +62,15 @@ async function createDataObject<TObj extends PureDataObject<O, S, E>, O, S, E ex
     // request mixin in
     runtimeClass = mixinRequestHandler(
         async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
-            (await PureDataObject.getDataObject(runtimeArg)).request(request),
+            (await PureDataObject.getDataObject(runtimeArg, existing)).request(request),
```

Comment:
I think we should avoid adding existing here.
Basically code below (end of this function) ensures that initialization is fully complete for create flow.
So any call to getDataObject() is done to fully initialized object, and thus existing it passes to finishInitialization should be always false.
getDataObject() is used in Bohemia (I think), so I'd try to keep signature the same.

Reply:
I can't find an elegant way of keeping this the same. I think we need to live with breaking this API.

The flow at the end of this function does indeed call this with `false`. However removing this in the mixin, will lead to breaking all other `existing` scenarios. I had clear manual reproes (outside of unit tests) with the clicker (for example) when the request processing with getdataobject would fail because the thing already exists (see `0x179`).

Reply:
I want to understand it better - let's chat.
Take a look at createOdspSearchDataStoreFactory in Bohemia - there is nowhere to get existing flag in that flow, unless we baske it into mixins (mixinRequestHandler, mixinSummaryHandler), i.e. existing is provided by mixin into mixin callback, not captured from ambient environment (createDataObject() closure).

Reply:
synced offline. Thanks, Vlad!